src/basic/util.h

```diff
@@ -193,7 +193,7 @@ static inline unsigned u64log2(uint64_t n) {
 
 static inline unsigned u32ctz(uint32_t n) {
 #if __SIZEOF_INT__ == 4
-        return __builtin_ctz(n);
+        return n ? __builtin_ctz(n) : 0;
```

Comment:
we usually avoid relying on C's downrgade-to-bool feature when we check values that are actually numbers and not bools or bool-like. i.e. we'd prefer if this was written like this:

```
return n != 0 ? __builtin_ctz(n) : 0;
```